metrics/src/recorder.rs

```diff
@@ -1,13 +1,105 @@
+use self::cell::RecorderOnceCell;
 use crate::{Counter, Gauge, Histogram, Key, KeyName, Unit};
 use core::fmt;
-use core::sync::atomic::{AtomicUsize, Ordering};
 
-static mut RECORDER: &'static dyn Recorder = &NoopRecorder;
```

Comment:
It is still required as the default for `recorder`. The `Option<T>` is only used internally because `once_cell`  uses it as well. I can trivially change it back to also store the `NoopRecorder` in the cell instead of the option.

Reply:
Ah yeah, true. I'll probably nuke `metrics::recorder()` to get rid of `NoopRecorder` but nothing that has to happen in this PR.